src/qt/window.cpp

```diff
@@ -287,6 +287,9 @@ void wxWindowQt::PostCreation(bool generic)
     SetForegroundColour(wxColour(GetHandle()->palette().foreground().color()));
 
     GetHandle()->setFont( wxWindowBase::GetFont().GetHandle() );
+
+    wxWindowCreateEvent event(this);
+    wxPostEvent(this, event);
```

Comment:
Do we really need to _post_ the event here asynchronously rather than _sending_ it via `HandleWindowEvent()`? This is inconsistent with the other ports and may result in subtle problems due to different event order, so unless there is some good reason not to handle this event immediately (in which case it really ought to be documented in a comment here), it should be changed.

Could you please check this? TIA!

Reply:
My rationale was that the derived class call PostCreation before calling their base class Create so I used wxPostEvent to ensure that object was fully constructed before clients received the event.

I'll have a look at moving the call and making it synchronous  and get back to you